lib/functions/rootfs/distro-specific.sh

```diff
@@ -188,22 +190,44 @@ function create_sources_list() {
 		chroot "${basedir}" /bin/bash -c "cat armbian.key | apt-key add -"
 	fi
 
+	declare -a components=()
+	if [[ "${when}" == "image" ]]; then # only include the 'main' component when deploying to image
```

Comment:
@igorpecovnik this part is, I think, more impacting. what do you think?

Reply:
This is wrong. We always want all components that exists. Upgrade from minimal image to desktop has to work without manipulating this file.

Reply:
Hmm, why? rootfs is not involved at all in upgrades.
The question here is that some packages that are listed in .packages or .additional files actually only reside in the repo. And if not correctly included, rootfs building will fail or include an older package.
The `main` repo logically (should) only contains kernel, uboot, bsp-cli.
While the `-desktop` and `-utils` repos actually contain said packages. For example: `system-monitoring-center` or `nala`. 
True/false?

Reply:
> rootfs is not involved at all in upgrades.

Let me rephrase my question. When I am generating a CLI image, how my .list will looks like?

Reply:
Exactly like before (complete, with main/-utils/-desktop).
What changes here is the .list used for building the _rootfs_, not the image.
For rootfs, this now only includes `-utils` and `-desktop`.
I'm doing analysis of the Armbian repos to find what impact this might have, but I'm gonna open a separate issue for this.

Reply:
Please see https://github.com/armbian/build/issues/5180

Reply:
> What changes here is the .list used for building the rootfs, not the image.

I am asking about the image which release is next week and this is changing entire mechanism on the head. Not saying its better but I am not 100% sure it will not break things apart. Fixes yes, features changes, no.

Reply:
You miss the point. The way it is, it's already broken.

Reply:
> You miss the point. The way it is, it's already broken.

I understand its broken, but now I don't understand anything as I don't have time to study and test changes in design!

Reply:
Don't worry, I'm sending a revert for this specific change. That way you can study it in isolation.

Reply:
Done. Sent draft #5183 revert of just this point, and the full repo analysis in #5180, for when you have the time.

Reply:
> for this specific change

Thanks. This will do.

I understand you want to do it better / right, but I don't want to gamble. We should be really just fixing small things, not redesign mechanism. We did what we could until start of the May.

> when you have the time.

My chunks of time are getting very small. I am afraid this and whole repository management will probably need to wait few weeks. There are other things I need to focus to, so we need to do something to engage more people here. It can't be just our show. I will try to use oras dump or use previous mechanism for getting packages. One should work, but I am afraid both could have troubles, perhaps those we have no power to control. I have pileup in the office / comm / private and will be few days on the road next week ...